app/liquid/liquid_tag_finder.rb

```diff
@@ -8,27 +21,71 @@ def template
     @template ||= Liquid::Template.parse(@content)
   end
 
+  # Returns the name of every plugin referenced in the liquid markup.
+  # For example, if there is any liquid tag (whether flow control or
+  # display) that references +plugins.petition[ref].fields+, then
+  # 'petition' will be in the list returned by this method.
   def plugin_names
     markups = all_liquid_tags.map{ |li| li.instance_values['markup'] }
     plugins = markups.map{|markup| markup.match(/plugins\.([a-zA-Z0-9_]+)/) }
     plugins.select(&:present?).map{|matches| matches[1]}.uniq
   end
 
+  # Returns the names of every partial referenced in the liquid markup.
+  # More examples can be found in the LiquidTagFinderSpec
+  # For example:
+  #     {% include 'thermometer', ref: 'modal' %}
+  # would return
+  #     ['thermometer']
   def partial_names
     all_include_tags.map{ |incl| partial_name_from_include(incl) }
   end
 
+  # Returns the names of every partial referenced in the liquid markup,
+  # along with the ref passed with it. More examples in the spec.
+  # Refs serve to allow multiple of one plugin on a page.
+  # For example:
+  #     {% include 'thermometer', ref: 'modal' %}
+  # would return
+  #     [['thermometer', 'modal']]
```

Comment:
Yes there is - each list item is a tuple of the plugin name and the ref. I'll updated the comment to be clearer.